go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/traefik/traefik/v2
 
-go 1.22
+go 1.23
```

Comment:
```suggestion
go 1.23.0
```

Reply:
I've checked various sources, starting with go1.23 releases notes, and it seems pretty common to use only major-minor in go.mod. I found a go blog post about [toolchain](https://go.dev/blog/toolchain) that is using this format, but even in this post, there is no specific format suggested.
Would you please detail this suggestion to fix version with this .0 ? 
Is this for using _toolchain_ introduced with go1.21 ?

Reply:
The format `goX.Y` is the release family/language version, this is not the same thing as the minimum Go version.

> At go 1.21 or higher:
> - The go line declares a required minimum version of Go to use with this module.
> - ...
>
> https://go.dev/ref/mod#go-mod-file-go

> Go versions
>
> Released versions of Go use the version syntax ‘1.N.P’, denoting the Pth release of Go 1.N. The initial release is 1.N.0, like in ‘1.21.0’. Later releases like 1.N.9 are often referred to as patch releases.
>
> Go 1.N release candidates, which are issued before 1.N.0, use the version syntax ‘1.NrcR’. The first release candidate for Go 1.N has version 1.Nrc1, like in 1.23rc1.
>
> The syntax ‘1.N’ is called a “language version”. It denotes the overall family of Go releases implementing that version of the Go language and standard library.
>
> The language version for a Go version is the result of truncating everything after the N: 1.21, 1.21rc2, and 1.21.3 all implement language version 1.21.
> https://go.dev/doc/toolchain#version

And [Code QL](https://codeql.github.com/) will report a warning:

![Screenshot 2024-07-05 at 16-03-09 Code scanning tool status · golangci_golangci-lint](https://github.com/gofrs/flock/assets/5674651/0055085c-9698-40d3-9458-363c3e1c68a9)